### PR TITLE
[WIP] Add Tests for Food & food/Name class

### PR DESCRIPTION
**Pull request by akhilvuputuri**

### CODE REVIEW

File: src/test/java/seedu/address/model/food/FoodTest.java [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+        assertTrue(PASTA.equals(PASTA));
+
+        // null -> returns false
+        assertFalse(PASTA.equals(null));
+
+        // different type -> returns false
+        assertFalse(PASTA.equals(5));
+
+        // different person -> returns false
+        assertFalse(PASTA.equals(BURGER));
```
Comment (sheexiong) [inline src/test/java/seedu/address/model/food/FoodTest.java]: I like the way you add the comment for each line to explain the expected behavior.

File: src/test/java/seedu/address/model/food/FoodTest.java [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+    public void equals() {
+        // same values -> returns true
+        Food pastaCopy = new FoodBuilder(PASTA).build();
+        assertTrue(PASTA.equals(pastaCopy));
+
+        // same object -> returns true
+        assertTrue(PASTA.equals(PASTA));
+
+        // null -> returns false
+        assertFalse(PASTA.equals(null));
```
Comment (sheexiong) [inline src/test/java/seedu/address/model/food/FoodTest.java]: If you use `equal()` to a Null, you will get `NullPointerException`. [Here](https://stackoverflow.com/questions/4501061) is the explanation from StackOverflow.

File: src/test/java/seedu/address/model/food/FoodTest.java [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+
+        // different type -> returns false
+        assertFalse(PASTA.equals(5));
+
+        // different person -> returns false
```
Comment (ailanthustng) [inline src/test/java/seedu/address/model/food/FoodTest.java]: I think you meant "different food -> returns false"

File: src/test/java/seedu/address/testutil/FoodBuilder.java [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+
+/**
+ * A utility class to help with building Food objects.
+ */
+public class FoodBuilder {
+
+    public static final String DEFAULT_NAME = "French Fries";
+    public static final long DEFAULT_CALORIES = 300;
+
+    private Name name;
```
Comment (ailanthustng) [inline src/test/java/seedu/address/testutil/FoodBuilder.java]: I recommend changing this to "FoodName" to make it clearer.

File: src/test/java/seedu/address/model/food/NameTest.java [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+        assertFalse(Name.isValidName(" ")); // spaces only
+        assertFalse(Name.isValidName("^")); // only non-alphanumeric characters
+        assertFalse(Name.isValidName("burger*")); // contains non-alphanumeric characters
+
+        // valid name
+        assertTrue(Name.isValidName("french fries")); // alphabets only
+        assertTrue(Name.isValidName("12345")); // numbers only
+        assertTrue(Name.isValidName("4fingers")); // alphanumeric characters
+        assertTrue(Name.isValidName("Aglio Olio")); // with capital letters
+        assertTrue(Name.isValidName("McSpicy Meal with Fries and Coke")); // long names
```
Comment (ailanthustng) [inline src/test/java/seedu/address/model/food/NameTest.java]: I like that you put the comments at the side to explain the validity and invalidity of the example Name.

File: src/test/java/seedu/address/testutil/FoodBuilder.java [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+    public FoodBuilder withTags(String ... tags) {
+        this.tags = SampleDataUtil.getTagSet(tags);
+        return this;
+    }
+
+    /**
+     * Builds a person.
+     *
+     * @return a Person.
+     */
```
Comment (ailanthustng) [inline src/test/java/seedu/address/testutil/FoodBuilder.java]: I think you meant Food here as well

File: src/test/java/seedu/address/testutil/FoodBuilder.java [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+     * Initializes the FoodBuilder with the data of {@code foodToCopy}.
+     */
+    public FoodBuilder(Food foodToCopy) {
+        name = foodToCopy.getName();
+        calories = foodToCopy.getCalories();
+        tags = new HashSet<>(foodToCopy.getTags());
+    }
+
+    /**
+     * Sets the {@code Name} of the {@code Person} that we are building.
```
Comment (ailanthustng) [inline src/test/java/seedu/address/testutil/FoodBuilder.java]: I think you meant Food

File: src/test/java/seedu/address/testutil/TypicalFood.java [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import seedu.address.model.FoodBook;
+import seedu.address.model.food.Food;
+
+/**
+ * A utility class containing a list of {@code Person} objects to be used in tests.
```
Comment (ailanthustng) [inline src/test/java/seedu/address/testutil/TypicalFood.java]: I think you meant Food

File: src/test/java/seedu/address/model/food/FoodTest.java [hunk 9]
```diff
@@ -1,4 +1,74 @@
...
+
+        // same name, same phone, different attributes -> returns true
+        editedAlice = new FoodBuilder(PASTA)
+                .withTags(VALID_TAG_HUSBAND).build();
+        assertTrue(PASTA.isSameFood(editedAlice));
+
+        // same name, same phone, same email, different attributes -> returns true
+        editedAlice = new FoodBuilder(PASTA).withTags(VALID_TAG_HUSBAND).build();
+        assertTrue(PASTA.isSameFood(editedAlice));
+    }
```
Comment (Jarrod-Bob) [inline src/test/java/seedu/address/model/food/FoodTest.java]: With regards to the comments, I think you may need to edit the comments as it seems to reflect a non-attributes of the Food Class in "same phone" and "same email".

File: src/test/java/seedu/address/testutil/FoodBuilder.java (hunk 4 above)
Comment (Jarrod-Bob) [reply src/test/java/seedu/address/testutil/FoodBuilder.java]: I think with our new project file organisation structure, this confusion is likely to be resolved by the abstractions of the 2 components (diet and splitter) into their own Logic, Model, etc.

**Review by Jarrod-Bob — COMMENTED** [review]
Generally well structured, although there should be caution exercised when using similar code from other components. You have to remember to change the comments for the functions, and not just the functions themselves, or it may be misleading for people who read our docs.

**Pull request closed without merging**